src/Serializers/BaseSerializer.ts

```diff
@@ -40,68 +42,76 @@ export abstract class BaseSerializer<S, T extends IModelData, R = T> implements
    * Returns a function, which when called, transforms the given fieldName on the provided model
    * into its serialized equivalent if the given IFieldType implements the optional "transform" method.
    *
-   * For example, a Date object will be converted into an ISO Date string when given a DateField.
+   * That function will then return a Promise that resolves to provide a tuple of the fieldName and the value.
+   * The Promise is necessary for the rare circumstance that we need to perform the transform asynchronously.
    *
-   * @param {IModel<T extends IModelData>} model
-   * @returns {(fieldType: IFieldType, fieldName: string) => IModel<T extends IModelData>[string]}
+   * For example, a Date object will be converted into an ISO Date string when given a DateField.
    */
   public transformField(model: IModel<T> | Partial<T>) {
-    return (fieldType: IFieldType & any, fieldName: string) => {
-      const fieldValue = model[fieldName];
-
-      if (fieldValue == null) {
-        return fieldValue;
-      }
+    return async (fieldType: IFieldType & any, fieldName: string): Promise<[string, string]> => {
+      let fieldValue = model[fieldName];
 
-      if (this.relationships && fieldName in this.relationships) {
-        return this.transformRelationship(fieldValue, this.relationships[fieldName]);
-      }
+      if (fieldValue != null) {
+        if (this.relationships && fieldName in this.relationships) {
+          fieldValue = await this.transformRelationship(fieldValue, this.relationships[fieldName]);
+        }
 
-      if ("transform" in fieldType) {
-        return fieldType.transform(fieldValue);
+        if ("transform" in fieldType) {
```

Comment:
This makes me a bit hesitant, but I guess that we can't really check if it's a certain type at runtime.  Curse you, Typescript.

Reply:
The check is because "transform" is optional on `IFieldType`, thus it might be undefined.

Reply:
It's the strings, man, it's the strings. 😄

Reply:
💯